
This implements gets (untested) for "minised".

  - Rene Rebe <rene@exactcode.de>

--- /dev/null	1970-01-01 01:00:00.000000000 +0100
+++ dietlibc-0.28/libstdio/gets.c	2005-02-02 16:41:10.406257872 +0100
@@ -0,0 +1,20 @@
+#include "dietstdio.h"
+#include "dietwarning.h"
+
+char *gets_unlocked(char *s) {
+  char *orig=s;
+  for (;;) {
+    register int c=fgetc_unlocked(stdin);
+    if (c==EOF) break;
+    *s=c;
+    ++s;
+    if (c=='\n') break;
+  }
+  if (ferror_unlocked(stdin))
+    return 0;
+  *s=0;
+  return orig;
+}
+
+char*gets(char*s) __attribute__((weak,alias("gets_unlocked")));
+link_warning("gets","warning: Avoid gets, tt is very insecure.")
